### PR TITLE
Update bazel version and dependencies

### PR DESCRIPTION
**Pull request by bnczk**
- Update bazel to 3.1.0
- Upgrade bazel-based deps to work with 3.1.0
- Upgrade jar deps to newest version to avoid conflict with bazel-based deps
- Update tests, removing deprecated methods and
- Remove unneeded deps

I also ran buildifier to migrate from the native {java,proto}_library to the skylark ones.
If there are too many changes in here, I can also split the PR into smaller ones.

### CODE REVIEW

File: src/main/proto/BUILD [hunk 1]
```diff
@@ -1,10 +1,14 @@
-load("@org_pubref_rules_protobuf//java:rules.bzl", "java_proto_library")
+load("@rules_proto//proto:defs.bzl", "proto_library")
+load("@rules_proto_grpc//java:defs.bzl", "java_grpc_library")
```
Comment (dinowernli) [inline src/main/proto/BUILD:2]: Any reason not to use the rules that come with the grpc_java?

(similar to here: https://github.com/grpc-ecosystem/polyglot/blob/43e31743dccbde8e61f4735a723fc975af1c687e/src/main/proto/BUILD#L3)

File: src/test/java/me/dinowernli/grpc/prometheus/integration/MonitoringClientInterceptorIntegrationTest.java [hunk 2]
```diff
@@ -22,10 +19,13 @@
 import org.junit.Before;
 import org.junit.Test;
 
+import java.io.IOException;
+
 import static com.google.common.truth.Truth.assertThat;
 
 /** Integration tests for the client-side monitoring pipeline. */
 public class MonitoringClientInterceptorIntegrationTest {
+  public static final String grpcServerName = "grpc-server";
```
Comment (dinowernli) [inline src/test/java/me/dinowernli/grpc/prometheus/integration/MonitoringClientInterceptorIntegrationTest.java:28]: nit: can this be private?

File: src/test/java/me/dinowernli/grpc/prometheus/integration/MonitoringServerInterceptorIntegrationTest.java [hunk 3]
```diff
@@ -29,13 +25,16 @@
...
+import java.io.IOException;
+
 import static com.google.common.truth.Truth.assertThat;
 
 /**
  * Integrations tests which make sure that if a service is started with a
  * {@link MonitoringServerInterceptor}, then all Prometheus metrics get recorded correctly.
  */
 public class MonitoringServerInterceptorIntegrationTest {
+  public static final String grpcServerName = "grpc-server";
```
Comment (dinowernli) [inline src/test/java/me/dinowernli/grpc/prometheus/integration/MonitoringServerInterceptorIntegrationTest.java:37]: ditto, can this be private?

**Comment by bnczk** [comment]
@googlebot I signed it!

File: src/main/proto/BUILD (hunk 1 above)
Comment (bnczk) [reply src/main/proto/BUILD]: I've simply replaced the stackb rules with more up to date ones and didn't think about the grpc-java ones. Updated in new version

**Comment by dinowernli** [comment]
Whoops, I accidentally lifted this out of draft mode, apologies. This look great, thanks for making all the changes.

I'm not sure why the travis build is not showing up here, but for posterity, this is the passing build: https://travis-ci.org/github/grpc-ecosystem/java-grpc-prometheus/jobs/693415531

@bnczk I'll wait for your last ack before merging given I ninja-graduated your PR from a draft :)

**Comment by bnczk** [comment]
No problem. If everything looks good, feel free to merge it. There are a couple of issues that can be probably closed afterwards

**Pull request merged**